### PR TITLE
[TESTMERGE PLEASE I BEG YOU] Meet the Mining Overseer

### PR DESCRIPTION
**Pull request by BobJoga**
### What and why
This PR does two things:

1. It increases the total mining positions from 3 to 5.
**Why:**
Miners die often and this might get worse with bobmed and the forbidden PR (you know which one), so this might be a necessity. Not only that, but our lavaland is probably the most extensive and filled to the brim with stuff, this lets other people experience the job without having to hog it on ready because well we have been quite populated  lately.

2. It adds a mining overseer job.
**Why:**
For the reasons stated above. Mining might get tougher than it already is - the overseer is less of a commander and more of a nurse, he is there not to mine but to provide support, be it repairing the mining base or healing fucked up miners. Not only that, but the overseer doubles (maybe triples, depending on what you consider) as a lavaland guard, with security radio access to prevent smuggling **illegal loot**.

### The overseer... glory to Cargotzka.

The overseer has access to the security radio channel, but no security equipment - he is meant to report, not to act on reports.

This pretty much sums up mining to 6 workers in total, assuming every slot is filled. Damn.

### The equipment

The overseer has his own small office in the lavaland base, and spawns with some healing stuff (medkit and normal hypospray) at the cost of the suit and mining voucher.

The overseer's office has 5 things of note:
An advanced automatic mining scanner.
A normal loaded defib, to save miners who have died just recently.
Three medkits (regular, brute, burn).
A single RCD to repair the mining base - no extra matter cartridges.
A prized PKA. It's literally just a normal PKA with 110% mod capacity.

### Preventing confusion
I have added a warning for people joining as overseer so they are aware of the purpose of the job.

image
*(I've fixed the on in typo)*

TLDR:
**They are not field workers unless the situation calls for it - overseers don't mine, they help their miners out.**

### The new mining command chain

The quartermaster still orders every supplyworker, but just as the warden is the vice-president of security, the overseer is the vice-president of supply. He orders the miners around, while being ordered around by the QM, so the QM can focus on the paper pushers instead. Pretty simple stuff.

## Why It's Good For The Game

This shakes up our job list with something slightly new, as well as more miners because having 3 powergamers in a round was not controversial enough.

## Changelog
:cl:
add: Added a mining overseer job, with a single slot. Papers, please.
add: Shaft miner now has 5 slots instead of 3.
/:cl:

### CODE REVIEW

**Comment by Echopsychotic** [comment]
Just sayin. and don't get me wrong I love the idea. But I think the blueshield change should be put in a different PR. Not the same one, or at least change it to let everyone know there is a blueshield change with this PR

**Comment by Feasel** [comment]
Can you move the Blueshield changes into their own PR? 

I feel like you're pulling a politician thing by hiding stuff in your PR's so you can get them in, and I don't want to be right.

Also, CC officials should get their /own/ Blueshields instead of stealing the one on station. The one on station is assigned to protect the heads of staff.

**Comment by BobJoga** [comment]
> Just sayin. and don't get me wrong I love the idea. But I think the blueshield change should be put in a different PR. Not the same one, or at least change it to let everyone know there is a blueshield change with this PR

Yes. Bad call putting it on this PR.

**Comment by BobJoga** [comment]
> Can you move the Blueshield changes into their own PR?
> 
> I feel like you're pulling a politician thing by hiding stuff in your PR's so you can get them in, and I don't want to be right.
> 
> Also, CC officials should get their /own/ Blueshields instead of stealing the one on station. The one on station is assigned to protect the heads of staff.

Yes i'm not arguing with that. I'll rectify this soon.

**Comment by SarmentiCampbell** [comment]
Do you believe it would be okay for the overseer to get a bag with surgeries tools?

**Comment by BobJoga** [comment]
> Do you believe it would be okay for the overseer to get a bag with surgeries tools?

I do but i want to avoid people throwing snakes at me, so i'd rather have them testmerged for a while alongg with Bobmed and other funny PRs to see how everything is received before i expand on them. Because mining is hated for various reasons and you know it.

**Comment by YakumoChen** [comment]
What's the point of having this guy start out with superior loot to the miners if he's not allowed to actually go out and use it?

**Comment by BobJoga** [comment]
> What's the point of having this guy start out with superior loot to the miners if he's not allowed to actually go out and use it?
> 
> Guarding the mining base isn't exactly a fun job like Warden (who has to deal with prisoners on top of being a guard). Rounds where the base is actually attacked are rare, and miners tend to not actually care since the environment is mostly hospitable and they have oxygen. Miners also tend to carry a lot of their own healing gear. Having someone whose job it is to go out and find bodies is useful but that's not a common enough occurrence to have someone dedicated full time babysitting the base.
> 
> What is the actual draw of this role? Miners spend the whole shift doing their own thing on the mine, I don't really see QMs interacting with them in the first place since they're self-sufficient, and I don't see a real reason to have someone who orders the miners around. I can't imagine they would listen too much anyway.
> 
> I could compare to brig doctor too but security is such a much more involved role with far more facets then "blast rocks, kill dudes" that it actually makes sense to have a brig doctor, not a mine doctor.

Well no. The brig doctor is literally just "heal preesoner, ERP with HoS". This job makes much more sense when you account the PRs i've mentioned here (#2578 and https://github.com/Skyrat-SS13/Skyrat13/pull/2945). When medicine is piss easy it is really hard for me to justify the medbay itself, The draw of this role is providing assistance for people that work in a hellish environment - there is no draw when the environment isn't really hellish. I don't know where you have taken the idea that he starts out with superior loot (he doesn't, it's just the PKA).

**Comment by CameronWoof** [comment]
I have a lot of trouble understanding what this PR adds to the server. I understand why someone would **want** to be a mining overseer, but it doesn't seem like it would be healthy for the server.

1. Mining already has a department head, and the Quartermaster has essentially no role or responsibility as it is. Taking the small amount of responsibility they have to ensure the miners are doing alright and shunting it off to another role doesn't seem like a great idea.

2. Advanced scanner? Special PKA? Three first-aid kits? Security radio? An RCD? The first two are for giving to your best metafriend in mining (since you're not allowed to use them yourself), the first-aid kit is one of the few things cargo can order that's actually useful, and why does someone with no security access or relation to the department need security radio?

This just seems like an antagbait role. It starts the round with everything you need to be an effective antagonist, all while its actual responsibilities are nebulous and actively infringe on what other roles (the QM and paramedics) should be doing.

I'd personally be much more interested in seeing a _medical_ job that's specialized in lavaland search and rescue than this.

**Comment by SarmentiCampbell** [comment]
From what I understand, the overseer is supposed to go rescue the corpses of miner who unfortunately died in lavaland, which is why they have a PKA to protect themself during it, they also have security radio to alert security about items like the cleaving saw who got stolen, I suppose? I do think it is a bad idea that they can be a antag though.

**Comment by CameronWoof** [comment]
I don't think security radio access is necessary to report crimes. The mining overseer shouldn't know about security issues and has no say in how law is enforced.

**Comment by BobJoga** [comment]
> I have a lot of trouble understanding what this PR adds to the server. I understand why someone would **want** to be a mining overseer, but it doesn't seem like it would be healthy for the server.
> 
> 1. Mining already has a department head, and the Quartermaster has essentially no role or responsibility as it is. Taking the small amount of responsibility they have to ensure the miners are doing alright and shunting it off to another role doesn't seem like a great idea.
> 2. Advanced scanner? Special PKA? Three first-aid kits? Security radio? An RCD? The first two are for giving to your best metafriend in mining (since you're not allowed to use them yourself), the first-aid kit is one of the few things cargo can order that's actually useful, and why does someone with no security access or relation to the department need security radio?
> 
> This just seems like an antagbait role. It starts the round with everything you need to be an effective antagonist, all while its actual responsibilities are nebulous and actively infringe on what other roles (the QM and paramedics) should be doing.
> 
> I'd personally be much more interested in seeing a _medical_ job that's specialized in lavaland search and rescue than this.

But this is the lavaland search and rescue job. The fuck do you want? I am more than willing to sacrifice some of their items even though i only really gave them the bare minimum.

**Comment by BobJoga** [comment]
> I don't think security radio access is necessary to report crimes. The mining overseer shouldn't know about security issues and has no say in how law is enforced.

I am fine with removing the access to the security radio then, but if the overseer becomes compliant to the issues miners bring and just doesn't report when miners do something shady, i pin the blame on you and no one else.

**Comment by CameronWoof** [comment]
It's my opinion that adding a sub-head role like this muddies the already really nebulous role of the Quartermaster. The main floor needs no oversight (one CT can hold the department alone), and miners are more or less self-governing. They don't need _leadership_ as much as they might need support.

I would suggest adding a clinic area to the lavaland mining station and adding this job as a Search-and-Rescue position (cargo/medical hybrid) rather than having it presented as a leadership role. If Miners are acting up, that's something the Quartermaster should be handling.

Again, this is just my opinion. Maintainers may feel differently. But if nothing else, I do think removing security radio access is a good step in the right direction.

**Comment by BobJoga** [comment]
> It's my opinion that adding a sub-head role like this muddies the already really nebulous role of the Quartermaster. The main floor needs no oversight (one CT can hold the department alone), and miners are more or less self-governing. They don't need _leadership_ as much as they might need support.
> 
> I would suggest adding a clinic area to the lavaland mining station and adding this job as a Search-and-Rescue position (cargo/medical hybrid) rather than having it presented as a leadership role. If Miners are acting up, that's something the Quartermaster should be handling.
> 
> Again, this is just my opinion. Maintainers may feel differently. But if nothing else, I do think removing security radio access is a good step in the right direction.

I am willing to remove the security radio. In fact i only added the radio so people would not complain about this being yet another miner powercreep thing and rather something to curb miner powercreep. I still feel the name of the role sounds cool and it being a subhead thing might be kept, but i will ask the opinions of the maintainers on this and adapt and overcome as it may come necessary.

**Comment by YakumoChen** [comment]
>Well no. The brig doctor is literally just "heal preesoner, ERP with HoS".

Disagree, brig doctor gets a lot of medical priority with security (who get hurt often) and prisoners (who hurt themselves often) and have an actual say on things related to sec and medical.

>The draw of this role is providing assistance for people that work in a hellish environment -

Miners take care of themselves. They have plenty of medical items they can both buy straight from the vendor and make on lavaland itself. It just seems more like a role to boss around miners who don't want or need it, even though the QM can already do that (and usually doesn't bother). 

>I don't know where you have taken the idea that he starts out with superior loot (he doesn't, it's just the PKA).

You get a better than average PKA that the miners would much rather prefer (110 means you can use three damage/cooldown mods) and the advanced mining scanner does literally nothing for a role that isn't allowed to mine, you just give it to one of the miners.

**Comment by BobJoga** [comment]
> Disagree, brig doctor gets a lot of medical priority with security (who get hurt often) and prisoners (who hurt themselves often) and have an actual say on things related to sec and medical.

They don't have any say in security matters and if you think they do you will eventually get bwoinked for it.

> Miners take care of themselves. They have plenty of medical items they can both buy straight from the vendor and make on lavaland itself. It just seems more like a role to boss around miners who don't want or need it, even though the QM can already do that (and usually doesn't bother). If the miners die, well then sucks. Most of the time miners can't even be revived easily because gib or chasm.

Again, you're entirely ignoring the PRs i mentioned which make mining an absolute unsustainable hell.

> You get a better than average PKA that the miners would much rather prefer (110 means you can use three damage/cooldown mods) and the advanced mining scanner does literally nothing for a role that isn't allowed to mine, you just give it to one of the miners.

Yes, you just give it to a miner. So what? Your job is helping miners. That is it.

**Comment by YakumoChen** [comment]
>They don't have any say in security matters and if you think they do you will eventually get bwoinked for it.

Any medical related issue in security is the brig medic's jurisdiction. That includes the well being and psych of prisoners and detainees, etc. It's stated outright in security policy.

> Again, you're entirely ignoring the PRs i mentioned which make mining an absolute unsustainable hell.

You are really overblowing it. People will take a couple weeks to get used to it at worst and it'll go back to the status quo once miners adapt.

> Yes, you just give it to a miner. So what? Your job is helping miners. That is it.

So you're basically admitting it's a glorified loot balloon who is stuck babysitting the mine the entire round.

**Comment by SarmentiCampbell** [comment]
I mean, do you forget that you literally fucking use it to protect yourself while also fetching dead miners? Only the advanced automatic mining scanner would be given to the miner if you didn't think that stupidly.

**Comment by YakumoChen** [comment]
Why give them a toy miners have to actually work for if it only exists to give the miners a free gimmie and doesn't benefit the user in the slightest?

**Comment by SarmentiCampbell** [comment]
...And?

**Comment by YakumoChen** [comment]
It's powercreep. Undeserved powercreep.

**Comment by SarmentiCampbell** [comment]
Honestly, At this point you're just saying that peoples will literally play as a overseer only to give the prized pka to miners, What? There will always be peoples who will hand out their equipments to others like the warden for example. 
Beside the overseer is literally to deal with miners related problems, literally go fetch their body while they get recloned, patch them up and send them back out or even literally help the miners deal with a megafauna.
Just because they can give out their pka which they problably shouldn't doesn't mean they are a bad role.

**Comment by BobJoga** [comment]
> Any medical related issue in security is the brig medic's jurisdiction. That includes the well being and psych of prisoners and detainees, etc. It's stated outright in security policy.

Boo hoo they have a job.

> > Again, you're entirely ignoring the PRs i mentioned which make mining an absolute unsustainable hell.
> 
> You are really overblowing it. People will take a couple weeks to get used to it at worst and it'll go back to the status quo once miners adapt. That's still ignoring the fact you can buy medkits from the vendor and treat yourself anyway.

No i'm not and i know this because as a frequent miner player i was fucking dying while testing both of those. If you truly believe miners can just take anything i throw at them, turning lavaland into kaizo hell, you are completely delusional and i won't hear anything you say.

> So you're basically admitting it's a glorified loot balloon (literally powercreep) who is stuck babysitting the mine the entire round.
Yes, whatever you say, at this point i'm just ignoring any comment you post from now on. Please keep shitposting and throwing shit at my PR.

balltze

**Comment by Ludox235** [comment]
**My opinion may seem absolutely useless and may be discarded (and I expect it to be) considering I don't fucking play here.**

But holy shit what an useless fucking role lmao.

So they get a better mining scanner and upgraded kinetic accelerator. Which is useless because they don't get to mine.
Their job is to babysit the miners. Which is useless since most miners die to megafauna/chasms, and they can just get healing items themselves.
Babysitting the miners is fucking boring. I have no idea how you think it will be fun to wait and hope a miner fucks up and dies.

They get a RCD. Woohoo. Not like miners can survive inside a completely fucked up base and that the RCD will be more useful given to miners so they can build ontop of lava.

Now, you may see that the only real use out of that role is to grab dead miners. (Since if a miner is injured they can just heal themselves)

So let's talk about that!

If a miner pre-scans it instantly nullifies the overseer's utility.
If the miner doesn't pre-scan and dies, pray to Shiva Gautama Christ-Chan that he has a GPS. And if he doesn't, then pray harder that he died to random fauna instead of megafauna or chasms (since gibbing would fuck up suit sensors.)

I don't know man, just give a doctor/paramedic mining access and boom you got your mining overseer that actually can be useful on station until a miner dies.

Overall the only "good" change was the miner slots getting upped to 5 to counter-act the absolute ultra-nightmare you're trying to make lavaland into.

**Again I don't expect this comment to be of any utility due to me not even playing here and that you basically called everyone in this PR delusional and retarded for saying that this idea was bad.**

_Do I have an idea why I bothered making this comment despite not playing here? No. Well kinda, it was the whole drama going on in the comments._

**Comment by BobJoga** [comment]
> **My opinion may seem absolutely useless and may be discarded (and I expect it to be) considering I don't fucking play here.**
> 
> But holy shit what an useless fucking role lmao.
> 
> So they get a better mining scanner and upgraded kinetic accelerator. Which is useless because they don't get to mine.
> Their job is to babysit the miners. Which is useless since most miners die to megafauna/chasms, and they can just get healing items themselves.
> Babysitting the miners is fucking boring. I have no idea how you think it will be fun to wait and hope a miner fucks up and dies.
> 
> They get a RCD. Woohoo. Not like miners can survive inside a completely fucked up base and that the RCD will be more useful given to miners so they can build ontop of lava.
> 
> Now, you may see that the only real use out of that role is to grab dead miners. (Since if a miner is injured they can just heal themselves)
> 
> So let's talk about that!
> 
> If a miner pre-scans it instantly nullifies the overseer's utility.
> If the miner doesn't pre-scan and dies, pray to Shiva Gautama Christ-Chan that he has a GPS. And if he doesn't, then pray harder that he died to random fauna instead of megafauna or chasms (since gibbing would fuck up suit sensors.)
> 
> I don't know man, just give a doctor/paramedic mining access and boom you got your mining overseer that actually can be useful on station until a miner dies.
> 
> Overall the only "good" change was the miner slots getting upped to 5 to counter-act the absolute ultra-nightmare you're trying to make lavaland into.
> 
> **Again I don't expect this comment to be of any utility due to me not even playing here and that you basically called everyone in this PR delusional and retarded for saying that this idea was bad.**
> 
> _Do I have an idea why I bothered making this comment despite not playing here? No. Well kinda, it was the whole drama going on in the comments._

I'm probably ditching this idea because it was just me going hmmmmm with a guy on discord and settle for more miner slots instead. Anyways thanks for your comment, it's as fruitless as you thought it would be. You raise some valid points but prescans will not come in cheap in the future so ehhh whatever. Good comment, i don't know you and i don't care to know you.

that said

> **Again I don't expect this comment to be of any utility due to me not even playing here and that you basically called everyone in this PR delusional and retarded for saying that this idea was bad.**

well no i only called one person here delusional but w/e and this idea is probably bad and i might just kind of throw it away

now go back to wherever you came from idk

**Comment by Jamini** [comment]
A few notes on Mining Overseer/Foreman

@BobJoga 

1. The regular KPA is more than enough to kill normal fauna. Frankly in my instances it's overkill. A mining overseer wouldn't need an extra edge at all unless they are really, really bad. Premium/Bloody KPAs are more upgrades for killing megafauna. I really don't see the point in giving this guy one.
2. In a similar vein, a regular mining scanner is fine. The advanced scanner is really good, and doesn't really serve any purpose in a role who isn't supposed to mine.
3. RCD is whatever. Those are really easy to get anyway.
4. I do not recommend giving them sec comms. Mining and Sci (just like a regular miner) would probably be enough. Crimes can be reported on the regular radio or handled lavaland style.
5.  If you DO insist on keeping security comms, I would recommend they be given gulag access and full control over the gulag prisoners. (Not like sec uses the gulag)

@SarmentiCampbell 

1. I see no reason why a mining overseer should get surgical tools. The role has nothing implying it is a full-scale medic. That seems more in the realm of proper search and rescue characters.

@Ludox235 

1. I could see one of my charcters doing mining overseer. I know of a few people (a lot of CHAPLAINS of all things) who would be good in this role.
2. Dying to megafauna doesn't necessarily mean unrecoverable.Smart miners stash their important gear and GPS to the side when fighting, and recovering gear is almost as important as recovering your corpse. Doubly so if your ID was in the pile.
3. Chasms don't permakill you here. Not sense Xenoarch was added.
4. Medical staff commonly have their hands full with actual doctor work. The only times I've seen a paramed do things on the surface, they kind of significantly hurt their actual department/job.
5. If you don't play here you probably shouldn't be commenting on our PRs. If only because _You seem to fundamentally misunderstand some of our existing systems._

**Comment by YakumoChen** [comment]
>Honestly, At this point you're just saying that peoples will literally play as a overseer only to give the prized pka to miners, What? There will always be peoples who will hand out their equipments to others like the warden for example.

The warden has tools that are actually useful to the warden. The gun he gets is optimal for the close quarters compact area of the brig, the gloves are better for handling prisoners than a baton, and the warden is always in the brig so the remote is more useful to him than an HoS that can be anywhere.

The overseer by contrast gets tools that are NOT useful to him and serve no purpose than to give to your favorite buddy miner for an early jump start.

Ludox hits it on the nose for the overseer covering imaginary problems that it isnt even well suited to deal with, and even useful applications of the role aren't considered or usable.
